node/pkg/driver/device_connectivity/device_connectivity_iscsi.go

```diff
@@ -97,22 +113,32 @@ func (r OsDeviceConnectivityIscsi) GetMpathDevice(volumeId string, lunId int, ar
 			   Return Value: "dm-X" of the volumeID by using the LunID and the arrayIqn.
 	*/
 	var devicePaths []string
-
+	var err error
 	lunIdStr := strconv.Itoa(lunId)
-	devicePath := strings.Join([]string{"/dev/disk/by-path/ip*", "iscsi", arrayIdentifier, "lun", lunIdStr}, "-")
-	klog.V(4).Infof("GetMpathDevice: Get the mpath devices related to arrayIdentifier=%s and lunID=%s : {%v}", arrayIdentifier, lunIdStr, devicePath)
 
-	devicePaths, exists, err := r.Helper.WaitForPathToExist(devicePath, 5, 1)
-	if err != nil {
-		klog.Errorf("GetMpathDevice: No device found error : %v ", err.Error())
-		return "", err
-	}
-	if !exists {
-		e := &MultipleDeviceNotFoundForLunError{volumeId, lunId, arrayIdentifier}
-		klog.Errorf(e.Error())
+	if len(arrayIdentifiers) == 0 {
+		e := &ErrorNotFoundArrayIdentifiers{lunId}
 		return "", e
 	}
 
+	for _, iqn := range arrayIdentifiers {
+		dp := strings.Join([]string{"/dev/disk/by-path/ip*", "iscsi", iqn, "lun", lunIdStr}, "-")
+		klog.V(4).Infof("GetMpathDevice: Get the mpath devices related to arrayIdentifier=%s and lunID=%s : {%v}", iqn, lunIdStr, dp)
+		dps, exists, e := r.Helper.WaitForPathToExist(dp, 5, 1)
+		if e != nil {
+			klog.Errorf("GetMpathDevice: No device found error : %v ", e.Error())
+			err = e
+		} else if !exists {
+			e := &MultipleDeviceNotFoundForLunError{volumeId, lunId, []string{iqn}}
+			klog.Errorf(e.Error())
+			err = e
+		}
+		devicePaths = append(devicePaths, dps...)
```

Comment:
If there is an error above, what happens here? does it still append the dps to the devicePaths?

Reply:
if there is an error, we will get nil as the dps here, devicePaths has not any change when appends nil

Reply:
If first time e != nil and second time exists == false, final err will be deviceNotFound, I don't think it is an expected err

Reply:
I will append all the error messages together